src/connectors/refinement-list/connectRefinementList.js

```diff
@@ -323,21 +305,12 @@ export default function connectRefinementList(renderFn, unmountFn) {
           [operator === 'and' ? 'facets' : 'disjunctiveFacets']: [attribute],
         };
 
-        if (limit !== undefined) {
-          const currentMaxValuesPerFacet = configuration.maxValuesPerFacet || 0;
-          if (showMoreLimit === undefined) {
-            widgetConfiguration.maxValuesPerFacet = Math.max(
-              currentMaxValuesPerFacet,
-              limit
-            );
-          } else {
-            widgetConfiguration.maxValuesPerFacet = Math.max(
-              currentMaxValuesPerFacet,
-              limit,
-              showMoreLimit
-            );
-          }
-        }
+        const currentMaxValuesPerFacet = configuration.maxValuesPerFacet || 0;
+
+        widgetConfiguration.maxValuesPerFacet = Math.max(
+          currentMaxValuesPerFacet,
+          showMore ? showMoreLimit : limit
+        );
```

Comment:
We can cover this with a test case. Like the one we have inside [`connectMenu`](https://github.com/algolia/instantsearch.js/blob/v3/src/connectors/menu/__tests__/connectMenu-test.js#L45-L61).